src/R/Editor/Impl/Completion/Providers/FilesCompletionProvider.cs

```diff
@@ -41,35 +49,42 @@ internal sealed class FilesCompletionProvider : IRCompletionListProvider {
             _glyphService = glyphService;
             _forceR = forceR;
 
-            _directory = ExtractDirectory(directoryCandidate);
-
-            if (_directory.Length == 0 || _directory.StartsWithOrdinal("~\\")) {
-                _directory = _directory.Length > 1 ? _directory.Substring(2) : _directory;
-                _userDirectoryFetchingTask = _workflow.RSession.GetRUserDirectoryAsync();
+            _enteredDirectory = ExtractDirectory(directoryCandidate);
+            if (_enteredDirectory.Length == 0 || _enteredDirectory.StartsWithOrdinal(".")) {
```

Comment:
I'd rather extract this whole part into separate method:
```cs
private async Task GetRootDirectory() {
    if (_enteredDirectory.Length == 0 || _enteredDirectory.StartsWithOrdinal(".")) {
        _rootDirectory = await _workflow.RSession.GetWorkingDirectoryAsync();
    } else if (_enteredDirectory.StartsWithOrdinal("~\\")) {
        _rootDirectory = await _workflow.RSession.GetRUserDirectoryAsync()
    }
}
```
```cs
_task = GetRootDirectory();
```

And we don't need to call `DoNotWait` on it, unless you want to fail fast in `FilesCompletionProvider` constructor on something.